softwareconfiguration/model/src/main/java/ai/libs/softwareconfiguration/optimizingfactory/OptimizingFactory.java

```diff
@@ -14,19 +14,19 @@
 
 import com.google.common.eventbus.Subscribe;
 
-import ai.libs.hasco.core.SoftwareConfigurationProblem;
-import ai.libs.hasco.exceptions.ComponentInstantiationFailedException;
-import ai.libs.hasco.model.ComponentInstance;
-import ai.libs.hasco.model.EvaluatedSoftwareConfigurationSolution;
 import ai.libs.jaicore.basic.algorithm.AAlgorithm;
 import ai.libs.jaicore.basic.algorithm.AlgorithmFinishedEvent;
 import ai.libs.jaicore.basic.algorithm.AlgorithmInitializedEvent;
 import ai.libs.jaicore.logging.ToJSONStringUtil;
+import ai.libs.softwareconfiguration.exceptions.ComponentInstantiationFailedException;
+import ai.libs.softwareconfiguration.model.ComponentInstance;
+import ai.libs.softwareconfiguration.model.EvaluatedSoftwareConfigurationSolution;
+import ai.libs.softwareconfiguration.model.SoftwareConfigurationProblem;
 
 public class OptimizingFactory<P extends SoftwareConfigurationProblem<V>, T, C extends EvaluatedSoftwareConfigurationSolution<V>, V extends Comparable<V>> extends AAlgorithm<OptimizingFactoryProblem<P, T, V>, T> {
 
 	/* logging */
-	private Logger logger = LoggerFactory.getLogger(OptimizingFactory.class);
+	private Logger localLogger = LoggerFactory.getLogger(OptimizingFactory.class);

```

Comment:
since when do we call loggers "localLogger"? this is rather confusing. **logger** for object loggers and **LOGGER** for class loggers.

Reply:
the parent class also has a logger, which is `protected`. This was a code smell, which is why I renamed it.

Reply:
Isnt then the better solution to make the parent's logger private?? I mean, why would you want to log something pretending to be your parent?

Reply:
This is indeed a tricky question. It could be nice to be able to use a logger out of the box other than setting up an individual logger each time. I mean you do not pretend to be the parent. You are just able use the same logger for the methods in both the parent class and your actual class.

Reply:
Of course, this takes away the possibility to configure log outputs of inherited methods and "own" methods differently.

Reply:
`AAlgorithm` is ILoggingCustomizable, and the idea was that you have, by default, a logger in your algorithm implementation (whenever you inherit from AAlgorithm). We have certainly not arrived at the best solution here. I think we should take this important discussion offline and also include the others, since this is an important design decision for the future.